app/lib/rocketchat/rocketchat.js

```diff
@@ -175,6 +175,17 @@ const RocketChat = {
 			message: I18n.t('Not_RC_Server', { contact: I18n.t('Contact_your_server_admin') })
 		};
 	},
+	async getServerTimeSync(server) {
+		try {
+			const response = await RNFetchBlob.fetch('GET', `${server}/_timesync`);
```

Comment:
I just realized we should do a race here.
This request might take some time to happen in some cases (poor connection fast enough to try to make the request, but not slow enough to be offline).
In such cases, it'd be better to just try for some time and show the lock screen than hanging for 10s.

I think a race of 2 seconds should be ok.
DM me if you're in doubt.